configs/module.go

```diff
@@ -179,9 +180,17 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 
 	for _, reqd := range file.RequiredProviders {
 		var fqn addrs.Provider
-		if reqd.Source != "" {
-			// FIXME: capture errors
-			fqn, _ = addrs.ParseProviderSourceString(reqd.Source)
+		if reqd.Source.SourceStr != "" {
+			var sourceDiags tfdiags.Diagnostics
+			fqn, sourceDiags = addrs.ParseProviderSourceString(reqd.Source.SourceStr)
+			if sourceDiags.HasErrors() {
```

Comment:
@apparentlymart I want to draw your attention here. You had made a comment in an earlier PR about a better method for transcribing this `tfdiags.Diagnostic` into an `hcl.Diagnostic`, but the only relevant method I could find translated only the source range to hcl (https://github.com/hashicorp/terraform/blob/master/tfdiags/hcl.go#L73).  

I appreciate any additional pointers/suggestions!

Reply:
Hmm yes it does seem like I was mistaken. I expect I was thinking of something elsewhere.

I think it would be technically possible to _create_ the thing I was describing, but that would be fine to do separately from this PR, and run with this localized solution for the moment.